src/schema/sources.ts

```diff
@@ -1075,20 +1076,26 @@ const togglePinnedPosts = async (
   return { _: true };
 };
 
-const getFormattedSources = async (entity, args, ctx): Promise<GQLSource[]> => {
+const getFormattedSources = async <Entity>(
+  entity: EntityTarget<Entity>,
+  args: { limit?: number },
+  ctx: Context,
+  info: GraphQLResolveInfo,
+): Promise<GQLSource[]> => {
   const { limit = 10 } = args;
-  /*
-   * This is not best practice, but due to missing join/basetable support for graphORM we decided to leave it like this to ship faster
-   * If you ever need child entities/graphORM magic this needs to change
-   */
-  return await ctx.con
-    .createQueryBuilder()
-    .select('s.*')
-    .from(entity, 'ts')
-    .innerJoin(Source, 's', 'ts."sourceId" = s.id')
-    .orderBy('r', 'DESC')
-    .limit(limit)
-    .getRawMany();
+
+  return await graphorm.query(ctx, info, (builder) => {
+    builder.queryBuilder
+      .innerJoin(
+        ctx.con.getMetadata(entity).tableName,
+        'ts',
+        `ts."sourceId" = ${builder.alias}.id`,
+      )
+      .where({ active: true, type: SourceType.Machine })
```

Comment:
I thought we didn't really care to showcase public squads here as well?

Reply:
It was by mistake 🙈